src/Link.cc

```diff
@@ -440,6 +440,65 @@ void Link::AddWorldForce(EntityComponentManager &_ecm,
   this->AddWorldWrench(_ecm, _force, torque);
 }
 
+//////////////////////////////////////////////////
+void Link::AddForceInInertialFrame(EntityComponentManager &_ecm,
+                                   const math::Vector3d &_force) const
+{
+  auto inertial = _ecm.Component<components::Inertial>(this->dataPtr->id);
+  auto worldPose = _ecm.ComponentData<components::WorldPose>(this->dataPtr->id)
+                       .value_or(sim::worldPose(this->dataPtr->id, _ecm));
+
+  // Can't apply force if the inertial's pose is not found
+  if (!inertial)
+    return;
+
+  // The force is expressed in terms of the link's inertial coordinate frame,
+  // We'll first convert this to force expressed
+  //in terms of link's coordinate frame
+  math::Vector3d linkForce = inertial->Data().Pose().Rot(
+                             ).RotateVector(_force);
+
+  // ExternalWorldForcecmd applies the force expressed in world coordinates
+  // so we need to compute the force expressed in world coordinates
+  math::Vector3d worldForce = worldPose.Rot().RotateVector(linkForce);
+
+  // Apply Force using AddWorldForce method
+  this->AddWorldForce(_ecm, worldForce);
+}
+
+//////////////////////////////////////////////////
+void Link::AddForceInInertialFrame(EntityComponentManager &_ecm,
+                                   const math::Vector3d &_force,
+                                   const math::Vector3d &_position) const
+{
+  auto inertial = _ecm.Component<components::Inertial>(this->dataPtr->id);
+  auto worldPose = _ecm.ComponentData<components::WorldPose>(this->dataPtr->id)
+                       .value_or(sim::worldPose(this->dataPtr->id, _ecm));
+
+  // Can't apply force if the inertial's pose is not found
+  if (!inertial)
+    return;
+
+  // The force is expressed in terms of the link's inertial coordinate frame,
+  // We'll first convert this to force expressed
+  //in terms of link's coordinate frame
+  math::Vector3d linkForce = inertial->Data().Pose().Rot(
+                             ).RotateVector(_force);
+
+  // ExternalWorldForcecmd applies the force expressed in world coordinates
+  // so we need to compute the force expressed in world coordinates
+  math::Vector3d worldForce = worldPose.Rot().RotateVector(linkForce);
+
+  // ExternalWorldForcecmd applies the force at a position expressed in terms
+  // of link's coordinate frame.So we compute the position expressed in terms
+  // of link's coordinate frame.
+  math ::Vector3d positionInLinkFrame = (_position - inertial->Data(
+                                         ).Pose().Pos());
```

Comment:
```suggestion
  math::Pose3d forceApplicationRelativeToInertialFrame(
      _position,
      math::Quaterniond::Identity);
  math::Vector3d positionInLinkFrame =
      (inertial->Data().Pose() *
       forceApplicationRelativeToInertialFrame).Pos();
```
`_position` is expressed in the inertial frame, so I think it needs to be rotated. Pose multiplication makes the most sense to me, and I think it would work properly as suggested here, though there is likely a more efficient code path. This code could be useful in the test even if not used in the implementation